src/core/processing/qgsprocessingutils.cpp

```diff
@@ -155,26 +155,30 @@ class ProjectionSettingRestorer
 
 QgsMapLayer *QgsProcessingUtils::loadMapLayerFromString( const QString &string )
 {
-  if ( QFileInfo::exists( string ) )
-  {
-    // TODO - remove when there is a cleaner way to block the unknown projection dialog!
-    ProjectionSettingRestorer restorer;
-    ( void )restorer; // no warnings
+  QStringList components = string.split( '|' );
+  if ( components.isEmpty() )
+    return nullptr;
 
-    QFileInfo fi( string );
-    QString name = fi.baseName();
+  if ( !QFileInfo::exists( components.at( 0 ) ) )
+    return nullptr;
 
-    // brute force attempt to load a matching layer
-    std::unique_ptr< QgsVectorLayer > layer( new QgsVectorLayer( string, name, QStringLiteral( "ogr" ), false ) );
-    if ( layer->isValid() )
-    {
-      return layer.release();
-    }
-    std::unique_ptr< QgsRasterLayer > rasterLayer( new QgsRasterLayer( string, name, QStringLiteral( "gdal" ), false ) );
-    if ( rasterLayer->isValid() )
-    {
-      return rasterLayer.release();
-    }
+  // TODO - remove when there is a cleaner way to block the unknown projection dialog!
+  ProjectionSettingRestorer restorer;
+  ( void )restorer; // no warnings
+
+  QFileInfo fi( components.at( 0 ) );
+  QString name = fi.baseName();
+
+  // brute force attempt to load a matching layer
+  std::unique_ptr< QgsVectorLayer > layer( new QgsVectorLayer( string, name, QStringLiteral( "ogr" ), false ) );
```

Comment:
@nyalldawson The `string` variable contains the full path and every parameters. The code is not updating this variable so it's not discarding the layerid or the subset.

Reply:
ah - i missed that!

Then the only risk here is if someone had a (valid?) path with a pipe character, but it looks like that's not possible on Windows and firmly discouraged on *nix too. I guess we could change the test to something like:

```
QFileInfo fi;
if ( QFileInfo::exists( string ) )
  fi = QFileInfo( string );
else if ( QFileInfo::exists( components.at( 0 ) ) )
  fi = QFileInfo( components.at( 0 )  );
else
  return nullptr;
```

Can you add a unit test for this too?